sdk/auth/managed_identity_authorizer.go

```diff
@@ -173,9 +177,9 @@ func azureMetadata(ctx context.Context, url string) (body []byte, err error) {
 		instanceMetadataService: true,
 
 		retryWaitMin:  2 * time.Second,
-		retryWaitMax:  60 * time.Second,
-		retryMaxCount: 5,
-		timeout:       10 * time.Second,
+		retryWaitMax:  retryWaitMax * time.Second,
+		retryMaxCount: retryMaxCount,
```

Comment:
these don't need to be configurable:

```suggestion
		retryWaitMax:  60 * time.Second,
		retryMaxCount: 5,
```